packages/sources/por-address-list/src/transport/multichainAddress.ts

```diff
@@ -0,0 +1,125 @@
+import { SubscriptionTransport } from '@chainlink/external-adapter-framework/transports/abstract/subscription'
+import { EndpointContext } from '@chainlink/external-adapter-framework/adapter'
+import { TransportDependencies } from '@chainlink/external-adapter-framework/transports'
+import { AdapterResponse, sleep } from '@chainlink/external-adapter-framework/util'
+import ABI from '../config/PoRAddressListMulti.json'
+import PolygonABI from '../config/MultiEVMPoRAddressList.json'
+import { BaseEndpointTypes, inputParameters } from '../endpoint/multichainAddress'
+import { ethers } from 'ethers'
+import { fetchAddressList, addProvider, getProvider } from './utils'
+
+export type AddressTransportTypes = BaseEndpointTypes
+
+type RequestParams = typeof inputParameters.validated
+
+interface ResponseSchema {
+  tokenSymbol: string
+  chain: string
+  chainId: bigint
+  tokenAddress: string
+  vaultAddress: string
+}
+
+export class AddressTransport extends SubscriptionTransport<AddressTransportTypes> {
+  providersMap: Record<string, ethers.providers.JsonRpcProvider> = {}
+  settings!: AddressTransportTypes['Settings']
+
+  async initialize(
+    dependencies: TransportDependencies<AddressTransportTypes>,
+    adapterSettings: AddressTransportTypes['Settings'],
+    endpointName: string,
+    transportName: string,
+  ): Promise<void> {
+    await super.initialize(dependencies, adapterSettings, endpointName, transportName)
+    this.settings = adapterSettings
+  }
+
+  async backgroundHandler(
+    context: EndpointContext<AddressTransportTypes>,
+    entries: RequestParams[],
+  ) {
+    await Promise.all(entries.map(async (param) => this.handleRequest(param)))
+    await sleep(context.adapterSettings.BACKGROUND_EXECUTE_MS)
+  }
+
+  async handleRequest(param: RequestParams) {
+    let response: AdapterResponse<BaseEndpointTypes['Response']>
+    try {
+      response = await this._handleRequest(param)
+    } catch (e) {
+      const errorMessage = e instanceof Error ? e.message : 'Unknown error occurred'
+      response = {
+        statusCode: 502,
+        errorMessage,
+        timestamps: {
+          providerDataRequestedUnixMs: 0,
+          providerDataReceivedUnixMs: 0,
+          providerIndicatedTimeUnixMs: undefined,
+        },
+      }
+    }
+    await this.responseCache.write(this.name, [{ params: param, response }])
+  }
+
+  async _handleRequest(
+    param: RequestParams,
+  ): Promise<AdapterResponse<AddressTransportTypes['Response']>> {
+    const { confirmations, contractAddress, contractAddressNetwork, batchSize } = param
+
+    this.providersMap = addProvider(contractAddressNetwork, this.providersMap)
+    const provider = getProvider(contractAddressNetwork, this.providersMap)
+
+    const addressManager = new ethers.Contract(
+      contractAddress,
+      contractAddressNetwork == 'POLYGON' ? PolygonABI : ABI,
```

Comment:
We are taking this as is given the timeline but Polygon need to re-do their contract to align with other ABIs later